incus/main.go

```diff
@@ -75,11 +75,21 @@ func listenAndServeTLS(conf incus.Configuration) {
 	}
 }
 
-func InstallSignalHandlers(signals chan os.Signal) {
+func InstallSignalHandlers() {
 	go func() {
+		signals := make(chan os.Signal, 1)
+		signal.Notify(signals, os.Interrupt, syscall.SIGTERM)
 		sig := <-signals
 		log.Printf("%v caught, incus is going down...", sig)
-		shutdown()
+		log.Printf("Waiting %d seconds for goroutines to shut down...", gracefulShutdownTimeout)
+
+		select {
+		case <-time.After(gracefulShutdownTimeout * time.Second):
+			shutdown()
```

Comment:
gives time for goroutines to exit. 

all of the goroutines are also listening for `os.Interrupt` and `syscall.SIGTERM`, so we give them time to gracefully close (longpoll messages show 503, websocket messages send a special close code indicating the server is going away (`1001 Going Away` [RFC 7.4.1](https://tools.ietf.org/html/rfc6455#section-7.4.1)))